internal/datastore/crdb/keys.go

```diff
@@ -49,3 +54,22 @@ func prefix(s string) string {
 	}
 	return prefix
 }
+
+// overlapKeysFromContext reads the request-provided initial overlap key set
+// from the grpc request metadata.
+func overlapKeysFromContext(ctx context.Context) keySet {
+	keys := newKeySet(ctx)
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return keys
+	}
+	for _, keyList := range md["spicedb-overlap"] {
+		for _, key := range strings.Split(strings.TrimSpace(keyList), ",") {
```

Comment:
why trim twice? if you're going to trim each individually, it's kinda pointless here right?